File: gateway/src/routes/health.ts
import { Hono } from "hono";

import type { Config } from "@/config.js";

export function healthRoute(config: Config): Hono {
  const app = new Hono();

  app.get("/health", (c) =>
    c.json({
      status: "ok",
      service: config.APP_NAME,
      env: config.NODE_ENV,
      version: "0.1.0",
    }),
  );

  return app;
}

File: backend/app/api/routes/health.py
from fastapi import APIRouter, Depends

from app.core.config import Settings, get_settings
from app.schemas.health import HealthResponse

router = APIRouter(tags=["system"])


@router.get("/health", response_model=HealthResponse)
def health(settings: Settings = Depends(get_settings)) -> HealthResponse:
    return HealthResponse(
        status="ok",
        service=settings.app_name,
        env=settings.app_env,
        version="0.1.0",
    )
